pkg/build/registry/buildconfig/etcd/etcd.go

```diff
@@ -47,3 +51,32 @@ func NewStorage(s storage.Interface) *REST {
 
 	return &REST{store}
 }
+
+func (r *REST) Delete(ctx kapi.Context, name string, options *kapi.DeleteOptions) (runtime.Object, error) {
+	bcObj, err := r.Get(ctx, name)
+	if err != nil {
+		return nil, err
+	}
+	bc := bcObj.(*api.BuildConfig)
+
+	// Just update DeletionTimestamp so Builds get deleted
+	if bc.DeletionTimestamp.IsZero() {
+		now := unversioned.Now()
+		bc.DeletionTimestamp = &now
+		bcObj, _, err = r.Update(ctx, bc)
+		return bcObj, err
```

Comment:
you're not really using this field as intended anymore, right? (this logic is no longer based on the upstream GracefulDelete path) so i think it would be better to introduce a new DeletePending boolean field, rather than abusing the DeletionTimestamp field which could result in unexpected behavior.